### ISSUE TITLE
freeflow mask defaults to constrained on backward expansion of time aware costmatrix request 

### ISSUE THREAD

**Issue by chrstnbwnkl** [valhalla/valhalla#4922]
With costmatrix supporting time aware requests, the user gets the benefit of getting predicted speeds on the forward search (with costmatrix currently only allowing `depart_at`), but not on the reverse search. If I follow `GraphTile::GetSpeed` correctly, the default speed source on the backwards search  is `kConstrainedFlow`. I wonder if there is a specific reason for this, it seems, well, a bit  constraining :smile: The issue I see is that there currently is no way to configure this without also affecting the speed_type config for the forward search, where you do have a valid time.

**Comment by kevinkreiser** [comment]
yeah it was on purpose. constrained speed is a good guess when you have no other idea about what the time of day is. constrained basically models what traffic is like when people are on the road which by definition would be the most likely time a person would want to run a route (when they are on the road). i could understand if the route depart_at time is at 9pm and the distance is short enough we could use freeflow instead (which is to model when no one is on the road).
